### DISCUSSION TITLE
What is the upload speed like for a single-node, single HDD setup in MinIO?

### DISCUSSION THREAD

**Discussion by adababys** [minio/minio#19449] [Q&A] [answered]
HDD 7200 RPM
Network: 1 Gbit/s 

Uploading a 200 KB file takes 300 milliseconds.
Uploading a 5 MB file takes 3000 milliseconds.

This seems too slow, is there a way to optimize it?

Thank you!

**Comment by klauspost** [comment, marked as answer]
Use SSD. We fsync writes, so you can't compare write speeds to your regular cached writes.

We aren't really doing anything magical, so you are just seeing the speed of the filesystem. We can't make magic happen there.

**Reply by adababys** [reply]
Thank you！I would like to ask about the performance comparison between MinIO using XFS and EXT4, BTRFS, or ZFS. Which one is better?

**Reply by klauspost** [reply]
XFS as we put in our documentation. EXT4 has limits that makes it problematic as a good backend.

**Comment by adababys** [comment]
Is there a relationship between the performance of MinIO and the number of nodes and the number of disks(HDD or SSD )?

**Reply by klauspost** [reply]
Yes.

**Comment by adababys** [comment]
> XFS as we put in our documentation. EXT4 has limits that makes it problematic as a good backend.

Could you please send me the document that describes the issues that arise when using EXT4 as backend storage?Thanks!

**Reply by klauspost** [reply]
@shaolila Reach out on `sales@min.io` with your inquiry.
